src/app/Http/Controllers/ApiController.php

```diff
@@ -126,12 +129,79 @@ public function updateAvailability(Request $request, Location $location) {
     public function armorvax(Request $request) {
         $data = $request->all();
         $data_string = json_encode($data, JSON_PRETTY_PRINT);
-        $path = sprintf('armorvax/%s/u%d_%s.json', config('app.env'), $request->user()->id, date('Y-m-d_his'));
+        $data_collection = collect(json_decode($data_string));
 
-        // Write the full JSON data to a file on S3
-        \Storage::disk('s3')->write($path, $data_string);
+        /**
+         * First things first: delete all the existing availabilities
+         * associated with the ArmorVax sites.
+         */
+        Location::locationsWithSource('ArmorVax')->each(
+        function ($location, $location_key) {
+            $location->availabilities->each(function ($availability, $key) {
+                # Delete the availabilities associated with each of the
+                # ArmorVax sites already in the database.
+                $availability->delete();
+            });
+        });
 
-        // echo back the same json that was posted
+        /**
+         * For each of the entries in the posted JSON array ...
+         */
+        $data_collection->each(
+            function ($location, $location_key) {
+                /**
+                 * ... For each of the locations in each of the entries ...
+                 */
+                collect($location->AvailibilitiesByLocation)->each(
+                    function ($availability, $availability_key) {
+                        $name = $availability->LocationName;
+                        $address = Address::standardize($availability->LocationAddress->AddressLine1);
+                        $address2 = Address::standardize($availability->LocationAddress->AddressLine2);
+                        $city = $availability->LocationAddress->City;
+                        $state = $availability->LocationAddress->State;
+                        $zip = $availability->LocationAddress->Zip;
+                        $county = $availability->LocationAddress->County;
+                        $booking_link = "https://app.armorvax.com";
+                        $source = LocationSource::firstOrCreate(['name' => 'ArmorVax'])->id;
+
+                        /**
+                         * ... Either find the location that matches or create a
+                         * new one ...
+                         */
+                        $location = Location::firstOrCreate([
```

Comment:
It looks like we're missing 'bookinglink' => $booking_link, in this array. You could also shorten with the compact function by changing your variable names above to $location_source_id and $bookinglink.

One other note about including all the fields in firstOrCreate... I believe this code would create a new location if there's even a single character different in any of the fields. I might suggest just matching on the address, and then we could update the other fields, considering ArmorVax the authoritative system. I think it would look something like this:

```$location = Location::firstOrCreate(compact(['address']));
$location->fill(compact([
'name',
'county',
'zip',
'state',
'address2',
'address',
'location_source_id',
'bookinglink',
]));
$location->save();
```

Reply:
I am wary of doing this. Let's continue to discuss after we come to a resolution on the other remaining topics!